### PR TITLE
feat: add automated release workflow

### PR DESCRIPTION
**Pull request by ayoung19**

### CODE REVIEW

**Review by Step7750 — CHANGES_REQUESTED** [review]
This is super cool! Thank you @ayoung19.

Few minor comments.

File: .github/workflows/release.yml [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+      - uses: actions/setup-node@v3
+        with:
+          node-version: ${{ matrix.node-version }}
+
+      - name: Build and compress
+        run: |
+          npm install
+          npm run build
+          zip -r extension-chrome.zip dist
+          npm run build_ff
```
[the remark this answers is not in the document]
Comment (ayoung19) [reply .github/workflows/release.yml:28]: great idea

File: .github/workflows/release.yml [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+      - name: Build and compress
+        run: |
+          npm install
+          npm run build
+          zip -r extension-chrome.zip dist
+          rm -rf dist
+          npm run build_ff
+          zip -r extension-firefox.zip dist
+
+      - uses: "marvinpinto/action-automatic-releases@latest"
```
Comment (Step7750) [inline .github/workflows/release.yml:32]: Let's hardcode this to use version `@v1.2.1`, we want to make this workflow deterministic when we can.

File: .github/workflows/upgrade.yml [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+          VERSION: ${{ inputs.version }}
+
+      - name: Add and commit
+        uses: EndBug/add-and-commit@v9
+        with:
+          message: "Bumps Version to v${{ inputs.version }}"
+          tag: "v${{ inputs.version }}"
+
+      - name: Push changes
+        uses: ad-m/github-push-action@master
```
Comment (Step7750) [inline .github/workflows/upgrade.yml:47]: Similarly hardcode to `@v0.6.0` for determinism.

**Review by Step7750 — APPROVED** [review]
Awesome stuff, thanks @ayoung19 !

**Pull request merged**